### PR TITLE
add support to run the tests on multi architecture 

### PR DESCRIPTION
**Pull request by jkandasa**
## Which problem is this PR solving?
- Currently e2e tests are not supported to run on non `x86_64` environment. due to lack of other dependent operator/images support
- Needs environment variables to filter required tests to run on a target platform. able to skip some of the tests which we can not run on the target platform at this time.

## Short description of the changes
- Introduced `SKIP_KAFKA` environment variable to skip kafka/streaming related tests. default: enabled
- Updated the code to react for `SKIP_ES_EXTERNAL` variable. In Openshift if `SKIP_ES_EXTERNAL` enabled, uses self provision Elasticsearch node.
- Updated to react for custom `VERTX_IMG` variable

### CODE REVIEW

**Review by frzifus — APPROVED** [review]

File: tests/e2e/Makefile [hunk 1]
```diff
@@ -11,6 +11,7 @@ export MULTI_ARCH_ASSERT_IMG ?= false
 USE_KIND_CLUSTER ?= true
 # Skip E2E tests where ES external instance is used
 SKIP_ES_EXTERNAL ?= false
+export SKIP_KAFKA ?= false
```
Comment (iblancasa) [inline tests/e2e/Makefile:14]: Add a comment in the same way there is one for `SKIP_ES_EXTERNAL`

File: tests/e2e/elasticsearch/render.sh [hunk 2]
```diff
@@ -8,16 +8,38 @@ if [ $IS_OPENSHIFT = true ]; then
 fi
 
 
+###############################################################################
+# TEST NAME: es-from-aio-to-production
+# DESCRIPTION: allInOne to production
```
Comment (iblancasa) [inline tests/e2e/elasticsearch/render.sh:13]: I think we can remove this

File: tests/e2e/examples/render.sh [hunk 3]
```diff
@@ -54,58 +77,100 @@ render_install_example "$example_name" "01"
...
+# TEST NAME: examples-with-cassandra
+###############################################################################
 start_test "examples-with-cassandra"
 example_name="with-cassandra"
 render_install_cassandra "00"
 render_install_example "$example_name" "01"
+# jaeger-cassandra-schema multi arch image is not available
+# created a PR: https://github.com/jaegertracing/jaeger/pull/4122 and https://github.com/jaegertracing/jaeger/pull/4123
+# until we get new release image, this is a workaround image
+# $YQ e -i '.spec.storage.cassandraCreateSchema.image = "quay.io/redhat-distributed-tracing-qe/jaeger-cassandra-schema:1.40.0"' ./01-install.yaml
```
Comment (iblancasa) [inline tests/e2e/examples/render.sh:166]: I don't get it. Do we need running this command or not?

**Review by iblancasa — CHANGES_REQUESTED** [review]

File: tests/e2e/elasticsearch/render.sh (hunk 2 above)
Comment (jkandasa) [reply tests/e2e/elasticsearch/render.sh:13]: @iblancasa with this header it is hard to identify the test

File: tests/e2e/examples/render.sh (hunk 3 above)
Comment (jkandasa) [reply tests/e2e/examples/render.sh:166]: @iblancasa sorry, we have to workaround. update the image on the operator. hence it is not required here. I will remove those lines

File: tests/e2e/elasticsearch/render.sh (hunk 2 above)
Comment (iblancasa) [reply tests/e2e/elasticsearch/render.sh:13]: I mean the `DESCRIPTION`

Comment (jkandasa) [reply tests/e2e/elasticsearch/render.sh:13]: ah ok, actually, I was confused about `aio`. then I looked into the steps and I got it `allInOne`. Hence I added this description :) 
if it looks odd, I can remove it.

**Review by iblancasa — APPROVED** [review]

**Review by frzifus — APPROVED** [review]

**Pull request merged**